src/plugindata.h

```diff
@@ -263,8 +265,66 @@ void plugin_configure_single(GtkWidget *parent);
 void plugin_help(void);
 void plugin_cleanup(void);
 
+/** Called by Geany when a plugin library is loaded.
+ *
+ * This is the original entry point. Implement and export this function to be loadable at all.
+ * Then fill in GeanyPlugin::info and GeanyPlugin::funcs of the passed @p plugin. Finally
+ * GEANY_PLUGIN_REGISTER() and specify a minimum supported API version.
+ *
+ * For all glory details please read @ref howto.
+ *
+ * @param plugin The unique plugin handle to your plugin. You must set some fields here.
+ * @param module The GModule associated with your plugin library.
+ * @param geany_api_version The API version exposed by the Geany instance that loads your plugin.
+ *           This probably not the same as the one your plugin was compiled against.
```

Comment:
Am 15. August 2015 19:28:26 MESZ, schrieb Matthew Brush <notifications@github.com>:
>> @@ -263,8 +265,66 @@ void plugin_configure_single(GtkWidget *parent);
>>  void plugin_help(void);
>>  void plugin_cleanup(void);
>>  
>> +/** Called by Geany when a plugin library is loaded.
>> + *
>> + * This is the original entry point. Implement and export this
>function to be loadable at all.
>> + * Then fill in GeanyPlugin::info and GeanyPlugin::funcs of the
>passed @p plugin. Finally
>> + * GEANY_PLUGIN_REGISTER() and specify a minimum supported API
>version.
>> + *
>> + * For all glory details please read @ref howto.
>> + *
>> + * @param plugin The unique plugin handle to your plugin. You must
>set some fields here.
>> + * @param module The GModule associated with your plugin library.
>> + * @param geany_api_version The API version exposed by the Geany
>instance that loads your plugin.
>> + *           This probably not the same as the one your plugin was
>compiled against.
>
>What is the intended purpose of this `geany_api_version` parameter? I
>looked in the demo and it seems not to be used. If it's not going to be
>used much/ever, couldn't we just add a separate function(s) like GTK+
>does to give the runtime version (ex. `gtk_get_*_version()` functions)?
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/geany/geany/pull/469/files#r37137520

It clearly depends on the plug-in how it's used. The intended purpose os to enable runtime/soft dependencies on geany APIs without having to bump the minimum supported version.

Yes, there could be a separate function, but the param works just as well and comes for free.